app/routes.py

```diff
@@ -1,2 +1,184 @@
-from flask import Blueprint
+from werkzeug.wrappers import ETagRequestMixin
+from app.models.goal import Goal
+from flask import Blueprint, request, make_response, jsonify
+from app import db
+from app.models.task import Task
+from sqlalchemy import asc, desc
+from datetime import datetime
+import requests
+import os
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["GET","POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort = request.args.get("sort")
+
+        if sort == "asc":
+            tasks = Task.query.order_by(asc("title"))
+        
+        elif sort == "desc":
+            tasks = Task.query.order_by(desc("title"))
+        
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.is_complete()
+            })
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        if "title" not in request_body.keys() or "description" not in request_body.keys() or "completed_at" not in request_body.keys():
+            return make_response({"details": "Invalid data"}, 400)
+
+        else:    
+            new_task = Task(title=request_body["title"], description=request_body["description"], completed_at=request_body["completed_at"])
+            
+            db.session.add(new_task)
+            db.session.commit()
+
+            return make_response({"task": new_task.make_json()}, 201)
+
+@tasks_bp.route("/<task_id>", methods= ["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
```

Comment:
We could use `get_or_404()` method here 
https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/#flask_sqlalchemy.BaseQuery.get_or_404